lib/src/metta/interpreter2.rs

```diff
@@ -468,7 +469,7 @@ mod tests {
     fn interpret_atom_evaluate_pure_expression_variable_name_conflict() {
         let space = space("(= (foo ($W)) True)");
         let result = interpret_atom(&space, atom("(eval (foo $W))", bind!{}));
-        assert_eq!(result[0].0, sym!("True"));
+        assert_eq!(result[0].0, metta_atom("True"));
```

Comment:
The previous test harness Tokenizer was a strange half-initialized Tokenizer, with some of the of the stdlib tokens present but not all of them.

I don’t see the value in that half-initialized tokenizer, and some of the other tests (types.rs in particular) depend on some of the stdlib tokens.  So I changed the test harness to use the tokenizer from a fully-initialized runner.  However, that means `True` and `False` became defined tokens.

If resolving to a symbol (instead of a GroundedAtom) is important for this test, the following code passes with either Tokenizer:

        let space = space("(= (foo ($W)) Yes)");
        let result = interpret_atom(&space, atom("(eval (foo $W))", bind!{}));
        assert_eq!(result[0].0, sym!("Yes"));

Alternatively, the 8d15e25 changes the types.rs tests to use their own version of the convenience functions, and changes the common test setup functions to use an empty Tokenizer.